mypy/checkexpr.py

```diff
@@ -6209,11 +6209,20 @@ class PolyTranslator(TypeTranslator):
     See docstring for apply_poly() for details.
     """
 
-    def __init__(self, poly_tvars: Sequence[TypeVarLikeType]) -> None:
+    def __init__(
+        self,
+        poly_tvars: Iterable[TypeVarLikeType],
+        bound_tvars: set[TypeVarLikeType] | None = None,
+        seen_aliases: set[TypeInfo] | None = None,
+    ) -> None:
         self.poly_tvars = set(poly_tvars)
         # This is a simplified version of TypeVarScope used during semantic analysis.
-        self.bound_tvars: set[TypeVarLikeType] = set()
-        self.seen_aliases: set[TypeInfo] = set()
+        if bound_tvars is None:
+            bound_tvars = set()
+        self.bound_tvars = bound_tvars
+        if seen_aliases is None:
+            seen_aliases = set()
+        self.seen_aliases = seen_aliases
```

Comment:
You could possibly do this slightly more simply using frozensets:

```diff
--- a/mypy/checkexpr.py
+++ b/mypy/checkexpr.py
@@ -6212,16 +6212,12 @@ class PolyTranslator(TypeTranslator):
     def __init__(
         self,
         poly_tvars: Iterable[TypeVarLikeType],
-        bound_tvars: set[TypeVarLikeType] | None = None,
-        seen_aliases: set[TypeInfo] | None = None,
+        bound_tvars: frozenset[TypeVarLikeType] = frozenset(),
+        seen_aliases: frozenset[TypeInfo] = frozenset(),
     ) -> None:
         self.poly_tvars = set(poly_tvars)
         # This is a simplified version of TypeVarScope used during semantic analysis.
-        if bound_tvars is None:
-            bound_tvars = set()
         self.bound_tvars = bound_tvars
-        if seen_aliases is None:
-            seen_aliases = set()
         self.seen_aliases = seen_aliases
```

The regression test you added passes with this change